### PR TITLE
chore: add telemetry reporting for packet methods

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

closes: #7437

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/04-channel/v2/keeper/msg_server.go [hunk 1]
```diff
@@ -226,6 +226,8 @@ func (k *Keeper) Acknowledgement(ctx context.Context, msg *types.MsgAcknowledgem
 		}
 	}
 
+	defer telemetry.ReportAcknowledgePacket(msg.Packet)
```
Comment (DimitrisJim) [inline modules/core/04-channel/v2/keeper/msg_server.go:229]: defer call already existed for recv

File: modules/core/internal/v2/telemetry/packet.go [hunk 2]
```diff
@@ -1,14 +1,56 @@
...
+func ReportRecvPacket(packet types.Packet) {
+	for _, payload := range packet.Payloads {
+		telemetry.IncrCounterWithLabels(
+			[]string{"tx", "msg", "ibc", types.EventTypeRecvPacket},
+			1,
+			[]metrics.Label{
+				telemetry.NewLabel(ibcmetrics.LabelSourcePort, payload.SourcePort),
+				telemetry.NewLabel(ibcmetrics.LabelSourceChannel, packet.SourceChannel),
+				telemetry.NewLabel(ibcmetrics.LabelDestinationPort, payload.DestinationPort),
+				telemetry.NewLabel(ibcmetrics.LabelDestinationChannel, packet.DestinationChannel),
```
Comment (DimitrisJim) [inline modules/core/internal/v2/telemetry/packet.go:21]: kept same labels here, unsure if more/less is needed.

Comment (gjermundgaraba) [reply modules/core/internal/v2/telemetry/packet.go:21]: I would imagine it's fine. If we need more, someone will probably tell us.

**Review by gjermundgaraba — APPROVED** [review]
Looks good to me! LGTM!

**Pull request merged**